src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs

```diff
@@ -192,8 +194,7 @@ public async Task AddSyntacticClassificationsAsync(Document document, ImmutableA
         if (root is null)
             return;
 
-        var classificationService = services.GetLanguageServices(root.Language).GetRequiredService<ISyntaxClassificationService>();
```

Comment:
service-retrieval continues to be surprising expensive when done in hot spots.  here we can trivially lift this out such that thsi value is provided in the constructor.